mlrun/run.py

```diff
@@ -718,35 +720,45 @@ def update_common(fn, spec):
         fn.metadata.categories = categories
         fn.metadata.labels = labels or fn.metadata.labels
 
-    def resolve_nuclio_subkind(kind):
-        is_nuclio = kind.startswith("nuclio")
-        subkind = kind[kind.find(":") + 1 :] if is_nuclio and ":" in kind else None
-        if kind == RuntimeKinds.serving:
-            is_nuclio = True
-            subkind = serving_subkind
-        return is_nuclio, subkind
+    def resolve_nuclio_sub_kind(_kind):
+        _is_nuclio = _kind.startswith("nuclio")
+        _sub_kind = (
+            _kind[_kind.find(":") + 1 :] if _is_nuclio and ":" in _kind else None
+        )
+        if _kind == RuntimeKinds.serving:
+            _is_nuclio = True
+            _sub_kind = serving_subkind
+        elif _kind == RuntimeKinds.application:
+            _is_nuclio = True
+        return _is_nuclio, _sub_kind
 
     if (
         not embed_code
         and not code_output
         and (not filename or filename.endswith(".ipynb"))
     ):
         raise ValueError(
-            "a valid code file must be specified "
+            "A valid code file must be specified "
             "when not using the embed_code option"
         )
 
     if kind == RuntimeKinds.databricks and not embed_code:
-        raise ValueError("databricks tasks only support embed_code=True")
+        raise ValueError("Databricks tasks only support embed_code=True")
+
+    if kind == RuntimeKinds.application:
+        if handler:
+            raise ValueError("Handler is not supported for application runtime")
+        filename = str(ApplicationRuntime.reverse_proxy_file_path)
+        handler = "Handler"
```

Comment:
```suggestion
        filename, handler = ApplicationRuntime.get_filename_and_handler()
```
It's better to ask the `ApplicationRuntime` class what these are - add a class method to the class to return them. This way any change in these will be local to the runtime class, and will not require searching through the code to find the reference here.